crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -542,8 +542,17 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
                 Flow::Remote { raw_event, .. } => Some(raw_event.clone()),
             };
 
+            let encryption_info = match &this.ctx.flow {
+                Flow::Local { .. } => None,
+                Flow::Remote { encryption_info, .. } => encryption_info.clone(),
+            };
+
             trace!("Applying edit");
-            Some(event_item.with_content(new_content, edit_json))
+            Some(
+                event_item
+                    .with_content(new_content, edit_json)
+                    .with_encryption_info(encryption_info),
```

Comment:
Do I understand correctly that this is the fix? If so, then we should rather:

1. extract the previous `kind`
2. replace the encryption info in there
3. use `with_kind` here to replace with the updated kind

Reply:
Well.. it seems I made a right mess out of all of this: I completely missunderstood what the `Kind` does. I thought it's supposed to hold the *original* event data with mutations coming in as changes to locally stored properties e.g. content. 

I went and reimplemented everything. Please have another look, hope it makes more sense now 🙏